aws/resource_aws_sns_platform_application_test.go

```diff
@@ -0,0 +1,447 @@
+							platformPrincipalCheck,
+						),
+					},
+					{
+						ResourceName:            resourceName,
+						ImportState:             true,
+						ImportStateVerify:       true,
+						ImportStateVerifyIgnore: []string{"platform_credential", "platform_principal"},
+					},
+				},
+			})
+		})
+	}
+}
+
+func TestAccAwsSnsPlatformApplication_basicAttributes(t *testing.T) {
+	platforms, err := testAccAwsSnsPlatformApplicationPlatformFromEnv()
+	if err != nil {
+		t.Skip(err)
+	}
+
+	resourceName := "aws_sns_platform_application.test"
+
+	var testCases = []struct {
+		AttributeKey         string
+		AttributeValue       string
+		AttributeValueUpdate string
+	}{
+		{
+			AttributeKey:         "success_feedback_sample_rate",
+			AttributeValue:       "100",
+			AttributeValueUpdate: "99",
+		},
+	}
+
+	for _, platform := range platforms {
+		t.Run(platform.Name, func(*testing.T) {
+			for _, tc := range testCases {
+				t.Run(fmt.Sprintf("%s/%s", platform.Name, tc.AttributeKey), func(*testing.T) {
+					name := fmt.Sprintf("tf-acc-%d", acctest.RandInt())
+
+					resource.Test(t, resource.TestCase{
+						PreCheck:     func() { testAccPreCheck(t) },
+						Providers:    testAccProviders,
+						CheckDestroy: testAccCheckAWSSNSPlatformApplicationDestroy,
+						Steps: []resource.TestStep{
+							{
+								Config: testAccAwsSnsPlatformApplicationConfig_basicAttribute(name, platform, tc.AttributeKey, tc.AttributeValue),
+								Check: resource.ComposeTestCheckFunc(
+									testAccCheckAwsSnsPlatformApplicationExists(resourceName),
+									resource.TestCheckResourceAttr(resourceName, tc.AttributeKey, tc.AttributeValue),
+								),
+							},
+							{
+								Config: testAccAwsSnsPlatformApplicationConfig_basicAttribute(name, platform, tc.AttributeKey, tc.AttributeValueUpdate),
+								Check: resource.ComposeTestCheckFunc(
+									testAccCheckAwsSnsPlatformApplicationExists(resourceName),
+									resource.TestCheckResourceAttr(resourceName, tc.AttributeKey, tc.AttributeValueUpdate),
+								),
+							},
+							{
+								ResourceName:            resourceName,
+								ImportState:             true,
+								ImportStateVerify:       true,
+								ImportStateVerifyIgnore: []string{"platform_credential", "platform_principal"},
+							},
+						},
+					})
+				})
+			}
+		})
+	}
+}
+
+func TestAccAwsSnsPlatformApplication_iamRoleAttributes(t *testing.T) {
+	platforms, err := testAccAwsSnsPlatformApplicationPlatformFromEnv()
+	if err != nil {
+		t.Skip(err)
+	}
+
+	resourceName := "aws_sns_platform_application.test"
+
+	var testCases = []string{
+		"failure_feedback_role_arn",
+		"success_feedback_role_arn",
+	}
+
+	for _, platform := range platforms {
+		t.Run(platform.Name, func(*testing.T) {
+			for _, tc := range testCases {
+				t.Run(fmt.Sprintf("%s/%s", platform.Name, tc), func(*testing.T) {
+					iamRoleName1 := fmt.Sprintf("tf-acc-%d", acctest.RandInt())
```

Comment:
Nitpick: Have you considered extracting the actual test logic into a function so it's slightly easier to read this?